File: 2190-most-frequent-number-following-key-in-an-array/2190-most-frequent-number-following-key-in-an-array.java
class Solution {
    public int mostFrequent(int[] nums, int key) {
           Map<Integer, Integer> counts = new HashMap<>();

        for (int i = 0; i < nums.length - 1; i++) {
            int numsKey = nums[i];
            int numsValue = nums[i + 1];
            if (numsKey == key) {
                counts.put(numsValue, counts.getOrDefault(numsValue, 0) + 1);
            }
        }

        int maxValue = 0;
        int maxCount = 0;
        System.out.println(counts);

        for (Map.Entry<Integer, Integer> entry : counts.entrySet()) {
            int specKey = entry.getKey();
            int specValue = entry.getValue();

            if (specValue > maxCount) {
                maxValue = specKey;
                maxCount = specValue;
            }
        }

        return maxValue;
    }
}

File: 2190-most-frequent-number-following-key-in-an-array/2190-most-frequent-number-following-key-in-an-array.py
class Solution(object):
    def mostFrequent(self, nums, key):
        """
        :type nums: List[int]
        :type key: int
        :rtype: int
        """
        freq = Counter(x for i, x in enumerate(nums) if i and nums[i-1] == key)
        print(freq)
        return max(freq, key=freq.get)